### PR TITLE
#98 show db metadata

### PR DESCRIPTION
**Pull request by BornaP**
Pull request regarding the #98 issue. 

I extracted table rendering from `database-list.jsx`. Tables, views, functions and stored procedures are now in a new component called `db-metadata-list.jsx`. At first I thought about having three separate components but than I realized there's a lot of shared code and it makes not much sense to keep them in separated files. 

By default, tables are shown when database loads, while others are not shown. All lists can be toggled (opened/closed) by a click. Component keeps information of opened/closed lists via its `state` object. Tables and views are clickable and by double-clicking on them default query is generated. For now, procedures and functions list items are not clickable, but maybe in future they could be - like for example showing SQL code on double clicking them. 

Of course, for this to work you have to use `sqlectron-core master` compiled, current version from npm won't work.

By working on this, I realized few more issues like when you click 'minus' icon to 'close' one database and double click another database, old ones remain 'open'. This is not a big bug but it's annoying when you want to change databases in the same connection. I opened new issue for this.
Also, when there are no tables in database nothing is shown. I think we should render metadata menu list containing a message that current db contains no tables or something like that (like here for views/functions and SP). Anyways, I'll open a new issue for this when PR is accepted, to keep this pull request clean from that.

Cheers,
Borna

### CODE REVIEW

File: src/renderer/components/db-metadata-list.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  toggleListCollapse(toggledList) {
+    const state = { ...this.state };
+    if (!state[toggledList]) {
+      state[toggledList] = true;
+    } else {
+      state[toggledList] = false;
+    }
+    this.setState(state);
```
Comment (maxcnunes) [inline src/renderer/components/db-metadata-list.jsx]: Could replace all this code with: 
```js
this.state({ [toggledList]: !this.state[toggledList] });
```

File: src/renderer/components/db-metadata-list.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    return listItems.map((item, idxChild) => {
+      if(listName === 'Tables' || listName === 'Views') {
+        return (
+          <span key={idxChild}
+            className="item list-item clickable"
+            onDoubleClick={() => onSelectTable(database, item)}>
+            {item.name}
+          </span>
+        );
+      } else {
```
Comment (maxcnunes) [inline src/renderer/components/db-metadata-list.jsx]: There is no need for `else` statement if the previous `if` is already leaving the function.

**Comment by maxcnunes** [comment]
Awesome pull request!! I'm gonna test it locally before merging.

Related to the found bugs/improvements. I agree it is better create separated issues for them.

**Comment by BornaP** [comment]
Thanks!

I agree completely with comments regarding the code. I'll make changes and push new commit!

File: src/renderer/components/db-metadata-list.jsx (hunk 1 above)
Comment (BornaP) [inline src/renderer/components/db-metadata-list.jsx]: Are you sure about this one though? Seems better to keep it this way and use native setState method than mutating this.state directly.

Comment (maxcnunes) [inline src/renderer/components/db-metadata-list.jsx]: Sorry. I mean: `this.setState({ [toggledList]: !this.state[toggledList] });` then there is no need for `if/else`.

**Comment by BornaP** [comment]
Changed! :)

**Comment by maxcnunes** [comment]
Thanks @BornaP :clap:

**Comment by BornaP** [comment]
You're welcome @maxcnunes !

**Pull request merged**